### PR TITLE
add a mqtt gateway plugin

### PR DESCRIPTION
**Pull request by atorber**
一个微信远程调用插件，使用MQTT连接本地或云端Wechaty客户端实现远程操作

https://www.yuque.com/atorber/chatflow/hwqf0tgqg9o51epc

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -59,11 +68,6 @@
     "url": "https://github.com/wechaty/plugin-contrib/issues"
   },
   "homepage": "https://github.com/wechaty/plugin-contrib#readme",
-  "git": {
```
Comment (huan) [inline package.json:62]: Why do you remove the git hook?

File: src/contrib/ding-dong/ding-dong.spec.ts [hunk 2]
```diff
@@ -54,7 +54,7 @@ test('isMatchConfig {mention: false}', async t => {
 
     const mentionMessage = await new Promise<Message>(resolve => {
       room.once('message', resolve)
-      fixture.mocker.player.say('ding', [fixture.mocker.bot]).to(fixture.mocker.room)
+      fixture.mocker.player.say('ding', [ fixture.mocker.bot ]).to(fixture.mocker.room)
```
Comment (huan) [inline src/contrib/ding-dong/ding-dong.spec.ts:57]: Please submit another PR to fix the linting, instead of including the lining fixes in this PR, to make this PR only related to the MQTT plugin.

File: tests/integration.spec.ts [hunk 3]
```diff
@@ -21,10 +21,15 @@ test('integration testing', async t => {
 
 test('plugin name', async t => {
   for (const plugin of Object.values(plugins)) {
+    console.info('plugin.name:', plugin)
```
Comment (huan) [inline tests/integration.spec.ts:24]: Remove the unnecessary `console.info`

**Review by huan — CHANGES_REQUESTED** [review]
Please submit the linting fixes in another PR.

This PR should only include the MQTT related code modifications.

File: package.json (hunk 1 above)
Comment (atorber) [reply package.json:62]: This is an error because I deleted it during submission and forgot to restore it afterwards.

File: src/contrib/ding-dong/ding-dong.spec.ts (hunk 2 above)
Comment (atorber) [reply src/contrib/ding-dong/ding-dong.spec.ts:57]: Okay, I will do it later and close this pull.
